src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/visits/VisitMigrationResource.kt

```diff
@@ -51,4 +62,59 @@ class VisitMigrationResource(private val visitsMigrationService: VisitsMigration
   )
   fun migrateVisits(@RequestBody @Valid migrationFilter: VisitsMigrationFilter) =
     visitsMigrationService.migrateVisits(migrationFilter)
+
+  @PreAuthorize("hasRole('ROLE_MIGRATE_VISITS')")
+  @GetMapping("/visits/history")
+  @Operation(
+    summary = "Lists all filtered migration history records un-paged for visits",
+    responses = [
+      ApiResponse(
+        responseCode = "200",
+        description = "All visit migration history records",
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Incorrect permissions to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      )
+    ]
+  )
+  suspend fun getAll(
+    @Parameter(
+      description = "Only include migrations started after this date time",
+      example = "2020-03-23T12:00:00",
+    )
+    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
+    @RequestParam fromDateTime: LocalDateTime? = null,
+
+    @Parameter(
+      description = "Only include migrations started before this date time",
+      example = "2020-03-24T12:00:00",
+    )
+    @DateTimeFormat(iso = DateTimeFormat.ISO.DATE_TIME)
+    @RequestParam toDateTime: LocalDateTime? = null,
+
+    @Parameter(
+      description = "When true only include migrations that had at least one failure",
+      example = "false",
+    ) @RequestParam includeOnlyFailures: Boolean = false,
+
+    @Parameter(
+      description = "Specify the prison associated with the migration",
+      example = "HEI",
+    ) @RequestParam prisonId: String? = null,
+  ) = migrationHistoryService.findAll(
+    HistoryFilter(
+      migrationTypes = listOf(MigrationType.VISITS.name),
+      fromDateTime = fromDateTime,
+      toDateTime = toDateTime,
+      includeOnlyFailures = includeOnlyFailures,
+      filterContains = prisonId?.let { """"prisonIds":["$it"]""" }
+    )
```

Comment:
can only query migrations that are done on a second prison, which is how the migration will be rolled out